xdsl/dialects/memref.py

```diff
@@ -484,20 +485,33 @@ def verify_(self) -> None:
                 "Global initial value is expected to be a "
                 "dense type or an unit attribute"
             )
+        if self.alignment is not None:
+            assert isa(self.alignment, IntegerAttr)
+            test = self.alignment.value.data
+            # test if is power of 2 with binary magic
+            if not ((test & (test - 1) == 0) and test != 0):
```

Comment:
Would you mind factoring this out into its own function called `is_power_of_two` or something? Also, I have a feeling that testing for `test.bit_count() == 1` might be marginally better but it's definitely a matter of style.

Reply:
I like your proposal better!